activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -592,6 +592,25 @@ def translate_exception_class(e, sql)
           exception
         end
 
+        def redact(sql)
+          return sql unless ActiveRecord::Base.redacted_sql_errors
+          print_redaction_warning if ActiveRecord::Base.connection.prepared_statements
+          if %w(SELECT UPDATE DELETE).any? { |type| sql.starts_with?(type) }
+            sql.gsub(/= ('.*'|TRUE|FALSE|\d*\.?\d*)/, "= <REDACTED>")
```

Comment:
That's a fair point. I kind of agree, but I wanted to see how far I could get parsing it with regexes. I've tested this against MySQL, SQLite, and PG, but it can all goes out the window if we're expected to filter raw SQL strings crafter by developers.